Detectors/Base/src/Detector.cxx

```diff
@@ -128,4 +129,22 @@ void Detector::defineLayerTurbo(Int_t nlay, Double_t phi0, Double_t r, Double_t
 {
 }
 
+void Detector::initFieldTrackingParams(int& integration, float& maxfield)
+{
+  auto vmc = TVirtualMC::GetMC();
+  auto field = vmc->GetMagField();
+  // set reasonable default values
+  integration = 2;
+  maxfield = 10;
+  // see if we can query the o2 field
+  if (dynamic_cast<o2::field::MagneticField*>(field) != nullptr) {
+    auto o2field = (o2::field::MagneticField*)field;
```

Comment:
of course a matter of taste, but why not
`auto o2field = dynamic_cast<o2::field::MagneticField*>(field);
if (o2field != nullptr) {`
otherwise you cast two times